### ISSUE TITLE
Unable to share saved search due to JavaScript errors: There was an error fetching a resource: Parser is unable to parse the response.

### ISSUE THREAD

**Issue by juliohm1978** [Graylog2/graylog2-server#7682]
I am unable to share saved searches as admin.

![image](https://user-images.githubusercontent.com/4145456/76533554-02cad100-6457-11ea-8226-8cc80b8dabfc.png)

The search dialog hangs on without response.

![image](https://user-images.githubusercontent.com/4145456/76533577-0ceccf80-6457-11ea-96eb-34ca347d767b.png)

In the browser debug console, the following error appears when clicking the Share button.

```
There was an error fetching a resource: Parser is unable to parse the response. Additional information: Not available
```

![image](https://user-images.githubusercontent.com/4145456/76534151-dbc0cf00-6457-11ea-9f96-d376618464e1.png)

I've tested using a Firefox private/anonymous window, so there's no cached resources. The error persists either way.

## Your Environment

* Graylog Version: graylog/graylog:3.2.3-1
* Installation via helm chart stable/graylog:1.6.1
* Elasticsearch Version: docker.elastic.co/elasticsearch/elasticsearch-oss:6.5.4
* MongoDB Version: unguiculus/mongodb-install:0.7
* Operating System: 

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
```
* Browser version: Firefox 73.0.1 (64-bit)

**Comment by dennisoelkers** [comment]
Hey @juliohm1978,
thanks for reporting this! When you are clicking on "Share search", your browser should perform a request to the backend. Can you take a look at the network tab and extract request/response from there, and add it to this ticket? That would help us tremendously debugging this issue.

**Comment by juliohm1978** [comment]
This is the entire screenshot of the developer tools

![image](https://user-images.githubusercontent.com/4145456/76535138-2b53ca80-6459-11ea-900e-979c01c70b8a.png)

I've cleared the history logs and clicked "Share search". No requests are made to the backend before the error.

**Comment by dennisoelkers** [comment]
You are filtering all requests for "JS", which means requests fetching javascript assets. You need to switch it over to "All" (right below the trash can icon).

**Comment by juliohm1978** [comment]
Sorry, my bad!

![image](https://user-images.githubusercontent.com/4145456/76535946-6c98aa00-645a-11ea-814d-74cfc9b69b96.png)

A request is made, but with a 404 response.

```
https://graylog.dev.tjpr.jus.br/api/views/5e692c423bcbeb00395208f0/share
```

**Comment by juliohm1978** [comment]
Funny enough, the request right after that gets a 200 OK.

```
https://graylog.dev.tjpr.jus.br/api/views/5e692c423bcbeb00395208ee/share/users
```

It returns a json response with a user list from our LDAP/AD.

**Comment by dennisoelkers** [comment]
I think the `404` is not the problem, because it is actually intended (no view sharing exists => 404). The error seems to be something parsing related, maybe invalid JSON. Can you share the response's payload? If not, can you check that it is valid JSON?

**Comment by juliohm1978** [comment]
Yes, the json returned for https://graylog.dev.tjpr.jus.br/api/views/5e692c423bcbeb00395208ee/share/users seems to be valid. This is a redacted version of the content:

```json
[{
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}, {
  "username": "NNN",
  "full_name": "NNN"
}]
```

I apologize for the redaction, but I'm not pasting usernames and full employee names from our corporate network.

The contents of both `username` and `full_name` seem to be valid strings, no special characters.

**Comment by juliohm1978** [comment]
I've been trying to debug through the developer tools, debugger and console in Firefox, but I can't seem to find the cause of the error. This stack trace is all that I have in the console, but it hardly points me to anything meaningful.

![image](https://user-images.githubusercontent.com/4145456/76803045-b9141a80-67b7-11ea-8039-7b01354cb208.png)

It seems, like you said, to have problems parsing a response from one of the AJAX queries.

![image](https://user-images.githubusercontent.com/4145456/76803179-f5e01180-67b7-11ea-93d2-45d681457e96.png)

But I can't find a way to make the debugger breakpoints show me the contents of the request.

**Comment by juliohm1978** [comment]
**Found a solution!**

I was able to get more details using Google Chrome developer tools.

The JavaScript that fails seems to get a good response from the server, but fails when calling `ServerAvailabilityActions.reportSuccess()`.

![image](https://user-images.githubusercontent.com/4145456/77235433-5b574800-6b94-11ea-93f0-4fa11d31fda7.png)

The developer tools shows this more detailed stacktrace.

![image](https://user-images.githubusercontent.com/4145456/77235882-6d3aea00-6b98-11ea-9a86-7fe951521a0d.png)

At the root of this stacktrace, the console shows the original error message:

![image](https://user-images.githubusercontent.com/4145456/77235923-a7a48700-6b98-11ea-96c7-1e900c3cc97e.png)

More specifically, `Parser is unable to parse the response` because `rawResponse` is `404 page not found`.

And that was the real catch! Turns out, the 404 response was partially responsible for the error. The cause was the reverse proxy from my cluster -- Nginx Ingress Controller. By default, it intercepts HTTP errors to give users a custom error page.

For anyone else struggling, you are using Graylog behind a reverse proxy, such as Nginx, make sure that HTTP errors are not intercepted. Make sure they pass through, otherwise the browser might get an HTML error page instead of a JSON response.

In Kubernetes, using an ingress nginx controller that's configured to intercept HTTP errors, that means using the annotation on the Graylog Ingress:

```yaml
nginx.ingress.kubernetes.io/configuration-snippet:
    proxy_intercept_errors off;
```

That resolved the issue, I'm now able to share my saved searches.

Best regards!
